installers/docker-compose/scripts/setup.sh

```diff
@@ -0,0 +1,144 @@
+
+# Generate and convert certificate
+# See https://www.digitalocean.com/community/tutorials/openssl-essentials-working-with-ssl-certificates-private-keys-and-csrs
+
+GITLAB_SERVICE_NAME=$(echo $(grep -v '^#' $PWD/../templates/.env | grep -e "GITLAB_SERVICE_NAME" | sed -e 's/.*=//'))
+
+CERT_CONFIG_FILE=$BUILD_DIR/cert.cnf
+[ -e $CERT_CONFIG_FILE ] && rm $CERT_CONFIG_FILE
+cp -r $PWD/../templates/cert.cnf $CERT_CONFIG_FILE
+sed -i 's/__HOST_DNS_NAME__/'$HOST_DNS_NAME'/g' $CERT_CONFIG_FILE
+sed -i 's/__GITLAB_SERVICE_NAME__/'$GITLAB_SERVICE_NAME'/g' $CERT_CONFIG_FILE
+sed -i 's/__HOST_DNS__/'$HOST_DNS'/g' $CERT_CONFIG_FILE
+
+openssl req \
+  -newkey rsa:2048 -nodes \
+  -keyout $HOST_KEY_FILE \
+  -x509 -days 365 \
+  -out $HOST_CERT_FILE \
+  -config $BUILD_DIR/cert.cnf \
+  -extensions req_ext \
+  -subj "/C=US/ST=NY/L=NY/O=XX/CN=$HOST_DNS_NAME"
+
+openssl x509 \
+  -in $HOST_CERT_FILE \
+  -outform der \
+  -out $HOST_DER_FILE
+
+# Generate Java trust store and register the certificate
+
+rm -rf $TRUST_STORE
+
+TRUSTSTORE_PASSWORD=$(echo $(grep -v '^#' $PWD/../templates/.env | grep -e "TRUSTSTORE_PASSWORD" | sed -e 's/.*=//'))
+
+keytool -import -noprompt -alias $HOST_DNS_NAME -keystore $TRUST_STORE -file $HOST_DER_FILE -storepass $TRUSTSTORE_PASSWORD
+
+# Copy certificates over for gitlab
+
+mkdir -p $BUILD_DIR/container-data/gitlab
+cp -r $BUILD_DIR/ssl $BUILD_DIR/container-data/gitlab
+
+##########################################
+# 	Generate .env from template
+##########################################
+
+# Generate .env file
+
+DOTENV_FILE=$BUILD_DIR/.env
+[ -e $DOTENV_FILE ] && rm $DOTENV_FILE
+cp -r $PWD/../templates/.env $DOTENV_FILE
+
+sed -i 's~__BUILD_DIR__~'$BUILD_DIR'~g' $DOTENV_FILE
+sed -i 's/__MONGO_PASSWORD__/'$MONGO_PASSWORD'/g' $DOTENV_FILE
+sed -i 's/__GITLAB_HOST__/'$HOST_DNS'/g' $DOTENV_FILE
+sed -i 's/__GITLAB_HOST_NAME__/'$HOST_DNS_NAME'/g' $DOTENV_FILE
+sed -i 's/__GITLAB_ROOT_PASSWORD__/'$GITLAB_ROOT_PASSWORD'/g' $DOTENV_FILE
+sed -i 's/__GITLAB_PRIVATE_ACCESS_TOKEN__/'$GITLAB_PRIVATE_ACCESS_TOKEN'/g' $DOTENV_FILE
+sed -i 's~__GITLAB_KEY_FILE__~'$GITLAB_KEY_FILE'~g' $DOTENV_FILE
+sed -i 's~__GITLAB_CERT_FILE__~'$GITLAB_CERT_FILE'~g' $DOTENV_FILE
+
+GITLAB_PORT=$(echo $(grep -v '^#' $DOTENV_FILE | grep -e "GITLAB_PORT" | sed -e 's/.*=//'))
+GITLAB_URL=https://$HOST_DNS:$GITLAB_PORT
+LEGEND_SDLC_PORT=$(echo $(grep -v '^#' $DOTENV_FILE | grep -e "LEGEND_SDLC_PORT" | sed -e 's/.*=//'))
+LEGEND_SDLC_PUBLIC_URL=http://$HOST_DNS:$LEGEND_SDLC_PORT
+LEGEND_ENGINE_PORT=$(echo $(grep -v '^#' $DOTENV_FILE | grep -e "LEGEND_ENGINE_PORT" | sed -e 's/.*=//'))
+LEGEND_ENGINE_PUBLIC_URL=http://$HOST_DNS:$LEGEND_ENGINE_PORT
+LEGEND_STUDIO_PORT=$(echo $(grep -v '^#' $DOTENV_FILE | grep -e "LEGEND_STUDIO_PORT" | sed -e 's/.*=//'))
+LEGEND_STUDIO_PUBLIC_URL=http://$HOST_DNS:$LEGEND_STUDIO_PORT
+
+sed -i 's~__GITLAB_URL__~'$GITLAB_URL'~g' $DOTENV_FILE
+sed -i 's~__LEGEND_SDLC_PUBLIC_URL__~'$LEGEND_SDLC_PUBLIC_URL'~g' $DOTENV_FILE
```

Comment:
Can we avoid duplication of the find/replaces ? We have them in the run-* scripts and here.

Reply:
hm, but each script does replace its own different set of things, I don't think having shared utils is great either, each script should know what to do with its template.